### PR TITLE
layouts/partials: add client-side pkg.go.dev redirection

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Updates #4.

### CODE REVIEW

**Comment by kortschak** [comment]
This works and given that it depends on JS, it won't interfere with `go get`. It pauses a bit on 404, but I think that's manageable.

File: layouts/404.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      "/v1/gonum/",
+      "/v1/plot/",
+      "/v1/netlib/",
+      "/v1/hdf5/",
+      "/v1/tools/",
+      "/v1/exp/"
+    ];
+    prefixes.forEach(prefix => {
+      if (window.location.href.startsWith(base+prefix) || window.location.href == trimTrailingSlash(base+prefix)) {
+          let docURL = window.location.href.replace(base, "https://pkg.go.dev/gonum.org")
```
Comment (sbinet) [inline layouts/404.html]: perhaps keep the protocol intact?
```suggestion
          let docURL = window.location.href.replace(base, window.location.protocol + "//pkg.go.dev/gonum.org")
```

Comment (kortschak) [reply layouts/404.html]: www.gonum.org always redirects to https, but OK.

**Comment by kortschak** [comment]
PTAL

**Review by sbinet — APPROVED** [review]

**Pull request merged**